code/drasil-code/lib/Language/Drasil/Code/Imperative/Modules.hs

```diff
@@ -266,7 +267,7 @@ genInputConstructor = do
         ctor <- genConstructor "InputParameters" cdesc (map pcAuto cparams)
           [block ics]
         return $ Just ctor
-  genCtor $ any (`elem` dl) ["get_input", "derived_values", 
+  genCtor $ any (`elem` dl) [showChsStr GetInput, "derived_values", 
```

Comment:
So this line makes me think that "derived_values" and "input_constraints" are of a similar nature to "get_input", and maybe should be dealt with at the same time. [Or maybe not, because things spin out of control if you do it that way. Investigation needed.]

Reply:
Noted, and will come back to this after the other feedback to this pull request has been resolved.